biz.aQute.bnd.remote.junit/src/aQute/bnd/remote/junit/JUnitFramework.java

```diff
@@ -311,45 +328,54 @@ public void report() throws InvalidSyntaxException {
 
 	/**
 	 * Show the installed bundles
+	 * 
+	 * @param out the output destination
+	 * @throws IllegalArgumentException if specified argument is {@code null}
 	 */
 	public void reportBundles(Appendable out) {
+		checkNonNullArgument(out, "Output destination instance cannot be null");
 		try (Formatter f = new Formatter(out)) {
 			Stream.of(framework.getBundleContext()
 				.getBundles())
-				.forEach(bb -> {
-					f.format("%4s %s\n", bundleStateToString(bb.getState()), bb);
-				});
+				.forEach(bb -> f.format("%4s %s%n", bundleStateToString(bb.getState()), bb));
 		}
 	}
 
 	/**
 	 * Show the registered service
+	 * 
+	 * @param out the output destination
+	 * @throws IllegalArgumentException if specified argument is {@code null}
 	 */
 	public void reportServices(Appendable out) throws InvalidSyntaxException {
+		checkNonNullArgument(out, "Output destination instance cannot be null");
 		try (Formatter f = new Formatter(out)) {
 			Stream.of(framework.getBundleContext()
-				.getAllServiceReferences(null, null))
-				.forEach(sref -> {
-					f.format("%s\n", sref);
-				});
+				.getServiceReferences((String) null, null))
+				.forEach(sref -> f.format("%s%n", sref));
 			f.flush();
 		}
 	}
 
 	/**
 	 * Wait for a Service Reference to be registered
 	 * 
-	 * @param class1 the name of the service
-	 * @param timeoutInMs the time to wait
+	 * @param clazz the name of the service
+	 * @param timeout the time to wait
 	 * @return a service reference
 	 * @throws InterruptedException
+	 * @throws IllegalArgumentException if any of the specified arguments is
+	 *             {@code null}
 	 */
-	public <T> Optional<ServiceReference<T>> waitForServiceReference(Class<T> class1, long timeoutInMs)
+	public <T> Optional<ServiceReference<T>> waitForServiceReference(Class<T> clazz, Duration timeout)
```

Comment:
What is the point of the change to use Duration when you only get the millis out of it anyway? Seems an unnecessary change.

Reply:
From the provider perspective, I agree with you but from consumer perspective, it is more flexible to use duration in their code and pass it directly to the methods accepting `Duration` as argument. As a consumer, you have the flexibility to use any type of duration and don't have to consider about millis at all.

Reply:
With TimeUnit, you can easily convert anything to millis.

```
TimeUnit.SECONDS.toMillis(5);
TimeUnit.MINUTES.toMillis(1);
```

Reply:
That also nicely avoids creating objects since it all long arithmetic.